### PR TITLE
Database schema diagram and entity relationship diagram

### PR DESCRIPTION
**Pull request by mpa-mxiang**
For this PR, we:

- Diagram a data model clearly and explain what it's doing.

image

- Create a file named schema_based_on_diagram.sql where you implement the database from the diagram.
- Join tables from many-to-many relationships might not appear in the diagram, but you still need them.
- Remember to add the FK indexes.

### CODE REVIEW

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @mpa-mxiang && @emarkees && @Ali-0111 🤝 🤝 🤝 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't contain the actual code that was required throughout this project, it looks like there was a misunderstanding of the project, would you please double-check the project requirements? You were supposed to write SQL code that implements the database as it is described in the schema figure given, but I saw you only submitted a drawing which was not the case.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Ali-0111** [comment]
@mpa-mxiang , Thanks,

**Comment by mpa-mxiang** [comment]
> @mpa-mxiang , Thanks,

No worries, hope this PR is detailed enough:)

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -1 +1,45 @@
...
+
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP,
+    patient_id INT,
+    status VARCHAR(30),
+    PRIMARY KEY(id),
+    FOREIGN KEY(patient_id) REFERENCES patients(id)
+);
+
```
Comment (IngDavidTM) [inline schema_based_on_diagram.sql:21]: - [x] Good job creating the main tables for your project :+1:, but as a project requirement all "join tables" for the many-to-many relationships should be created, even if they are not shown on the diagram. Therefore you should create a table between medical_histories and treatments  (name can be different):

**Hint:**
```sql
  CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

File: schema_based_on_diagram.sql
Comment (IngDavidTM) [inline schema_based_on_diagram.sql:1]: - [x] Kindly don't forget that all FOREIGN KEYS should be set (by using `refrences table_name(id)` in column definition):

* medical_history_id in invoices
* medical_history_id in medical_histories_has_treatments
* treatment_id in medical_histories_has_treatments

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -1 +1,45 @@
...
+  unite_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  PRIMARY KEY(id),
+  FOREIGN KEY(invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY(treatment_id) REFERENCES treatments(id),
+);
+
```
Comment (IngDavidTM) [inline schema_based_on_diagram.sql]: - [x] Finally please don't forget that all FOREIGN KEYS should have indexes created:

* `CREATE INDEX ON medical_histories (patient_id);`
* `CREATE INDEX ON invoices (medical_history_id);`
* `CREATE INDEX ON invoice_items (invoice_id);`
* `CREATE INDEX ON invoice_items (treatment_id);`
* `CREATE INDEX ON medical_histories_has_treatments (medical_history_id);`
* `CREATE INDEX ON medical_histories_has_treatments (treatment_id);`

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @mpa-mxiang  🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @mpa-mxiang 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Tables created properly 👍 
* Good looking readme 👍 
* Tables joined properly 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mpa-mxiang** [comment]
Thank you @IngDavidTM  for the hints and thank you @Davidosky007 for the approval!

**Pull request merged**